src/types.rs

```diff
@@ -258,73 +260,122 @@ fn map_address_from_binary(v: &Option<FixedSizeData<20>>, should_checksum: bool)
     })
 }
 
-fn map_binary<T: Hex>(v: &Option<T>) -> Option<String> {
+fn map_hex_string<T: Hex>(v: &Option<T>) -> Option<String> {
     v.as_ref().map(|v| v.encode_hex())
 }
 
+pub(crate) fn map_opt_result<T, V, F>(v: Option<T>, f: F) -> Result<Option<V>>
+where
+    F: Fn(T) -> Result<V>,
+{
+    match v {
+        Some(v) => Ok(Some(f(v)?)),
+        None => Ok(None),
+    }
+}
+
+fn map_i64_from_u64(v: Option<u64>) -> Result<Option<i64>> {
+    map_opt_result(v, |v| i64::try_from(v).context("converting u64 to i64"))
+}
+
+fn map_i64<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<i64>> {
+    map_opt_result(v.as_ref(), |v| {
+        v.as_ref()
+            .try_into()
+            .context("converting to fixed size bytes i64")
+            .map(i64::from_be_bytes)
```

Comment:
Oh one other thing can you confirm we are using big endian bytes?

Reply:
can use ruint u256 type and the conversion functions implemented in code already, wrote details in other comment